packages/backend/test/peripherals/database/TransactionCountRepository.test.ts

```diff
@@ -0,0 +1,117 @@
+import { Logger } from '@l2beat/common'
+import { ProjectId, UnixTime } from '@l2beat/types'
+import { expect } from 'earljs'
+
+import {
+  TransactionCountRecord,
+  TransactionCountRepository,
+} from '../../../src/peripherals/database/TransactionCountRepository'
+import { setupDatabaseTestSuite } from './shared/setup'
+
+describe(TransactionCountRepository.name, () => {
+  const { database } = setupDatabaseTestSuite()
+  const repository = new TransactionCountRepository(database, Logger.SILENT)
+
+  const PROJECT_A = ProjectId('project-a')
+  const PROJECT_B = ProjectId('project-b')
+
+  beforeEach(async () => {
+    await repository.deleteAll()
+  })
+
+  describe(
+    TransactionCountRepository.prototype.getMissingRangesByProject.name,
+    () => {
+      it('finds holes', async () => {
```

Comment:
There is still an edge case test missing for a completely empty repository. We want it to return [-Infinity, Infinity]